plugin.rb

```diff
@@ -20,63 +20,80 @@ class Engine < ::Rails::Engine
   require_dependency "application_controller"
 
   class DiscourseSitemap::SitemapController < ::ApplicationController
-    def generate
-      prepend_view_path "plugins/discourse-sitemapcrap/app/views/" 
-      # this code is never called somehow
-    end
-  end
+    layout false
+    skip_before_filter :preload_json, :check_xhr, :redirect_to_login_if_required
```

Comment:
[RobotsTxtController](https://github.com/discourse/discourse/blob/master/app/controllers/robots_txt_controller.rb#L3) skipping the `redirect_to_login_if_required` filter. And `sitemap.xml` file's reference included in `robots.txt` file. So after looking into `robots.txt` file automatically bots will be looking for `sitemap.xml` file I guess.

In this case still I can remove it? I am not sure about how bot things will work.